xpdacq/xpdacq.py

```diff
@@ -192,6 +192,58 @@ def _parse_calibration_file(config_file_name):
                 config_dict[option] = None
     return config_dict
 
+def new_prun(sample, scanplan, auto_dark = glbl.auto_dark, **kwargs):
+    '''on this 'sample' run this 'scanplan'
+        
+    Arguments:
+    sample - sample metadata object
+    scanplan - scanplan metadata object
+    auto_dark - optional. Type auto_dark = False to suppress the automatic collection of a dark image (default = True). Strongly recommend to leave as True unless problems are encountered
+    **kwargs - dictionary that will be passed through to the run-engine metadata
+    '''
+    scan = Scan(sample, scanplan)
+    scan.md.update({'sc_isprun':True})
+    if auto_dark:
+        auto_dark_md_dict = _auto_dark(scan)
```

Comment:
OK, sounds good.

I also wonder whether auto_dark should just directly update scan.md rather
than passing back the new dictionary which is then added to ``scan.md`` in
``prun``.  The advantage of the former is that there is less duplicated
code (in background, setupscan and so on).  I am not sure if it will work
though....whether scan i s global enough....

S


On Mon, Mar 21, 2016 at 10:00 AM, Timothy Liu <notifications@github.com>
wrote:

> In xpdacq/xpdacq.py
> <https://github.com/xpdAcq/xpdAcq/pull/156#discussion_r56825561>:
>
> > @@ -192,6 +192,58 @@ def _parse_calibration_file(config_file_name):
> >                  config_dict[option] = None
> >      return config_dict
> >
> > +def new_prun(sample, scanplan, auto_dark = glbl.auto_dark, **kwargs):
> > +    '''on this 'sample' run this 'scanplan'
> > +
> > +    Arguments:
> > +    sample - sample metadata object
> > +    scanplan - scanplan metadata object
> > +    auto_dark - optional. Type auto_dark = False to suppress the automatic collection of a dark image (default = True). Strongly recommend to leave as True unless problems are encountered
> > +    **kwargs - dictionary that will be passed through to the run-engine metadata
> > +    '''
> > +    scan = Scan(sample, scanplan)
> > +    scan.md.update({'sc_isprun':True})
> > +    if auto_dark:
> > +        auto_dark_md_dict = _auto_dark(scan)
>
> Do you mean auto_dark and _auto_dark ? _auto_dark is a function that
> executes automated dark collection and auto_dark is an option here. I
> think I'd better change function name cause it is even very confusing for
> me by just looking at it.
>
> I am still working on this PR so I will correct it in following commits.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/xpdAcq/xpdAcq/pull/156/files/b5cfa3989d4cbf78f9cbca5c529e98a22bbab4ac#r56825561>
>



-- 
Prof. Simon Billinge
Applied Physics & Applied Mathematics
Columbia University
500 West 120th Street
Room 200 Mudd, MC 4701
New York, NY 10027
Tel: (212)-854-2918 (o) 851-7428 (lab)

Condensed Matter Physics and Materials Science Dept.
Brookhaven National Laboratory
P.O. Box 5000
Upton, NY 11973-5000
(631)-344-5661

email: sb2896 at columbia dot edu
home: http:// <http://nirt.pa.msu.edu/>bgsite.apam.columbia.edu/